docs/adr/0002-release-and-branching-strategy.md

```diff
@@ -0,0 +1,208 @@
+  * Good, because it scales well for a large number of collaborators.
+  * Good, because it is foundational to a variety of more tailored branching strategies.
+  * Bad, because it is slightly more intricate than the [Default](#default) strategy.
+  * Bad, because it involves an extra branch (`develop`) meaning more PRs and merges are necessary to manage code and releases.
+  * Bad, because the generic Gitflow strategy does not cover *all* development or release needs, and is generally designed for extension, so it is a common trap to feel the need to contort in order to remain within the bounds of generic Gitflow guidelines.
+  * Bad, because it does not provide out-of-the-box guidance for navigating development dependencies, and may not address the [problem statement](#context-and-problem-statement) without changes to the release process as well.
+
+#### [Modified Gitflow](https://docs.google.com/document/d/1FGW1u3W6T1M-HSw8Jw3xcjM_8HdnOsV5rdbYphPuLxY/edit#)
+* Characteristics
+  * An extension of the [default branching strategy](#default) via tailored additions of [Gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) concepts to match development and release needs of the `react-uswds` project.
+  * `main` is the primary develop branch.
+  * Features are PRed to `main`.
+  * Bugfixes are PRed to `main`.
+  * Additional develop branches are the primary extension of the default branching strategy
+  * Releases take place via short-lived release branches based off `main`, tagged on `main`.
+    * As noted below, this strategy also supports custom releases from the additional develop branches, if desired.
+* Pros and Cons
+  * Good, because it scales extremely well to *both* custom developer workflows and custom release strategies
+    * During times of low developer availability or periods without development dependencies the branching strategy can be scaled to be effectively identical to the [Default](#default) strategy.
+    * Its flexibility supports multiple release strategies, ideal for future changes to that process.
+  * Good, because it provides a clear process for navigating development dependencies and unblocking corresponding work.
+  * Good, because it retains the familiar process of PRing off of and to a base develop branch.
+  * Bad because, while based on well documented and widely understood concepts, this custom branching strategy would require additional documentation to make clear for new contributors.
+  * Bad because, it increases developer overhead by adding more considerations to the development process (i.e. "What branch do I use as a base for the work I want to do?").
+    * Bad, because it functions best with supporting automation which would likely be a custom development in this case.
+  * Bad because, and this is a *big* one: existing CI and merge processes would need to be re-evaluated.
```

Comment:
I think we should have a defined branch prefix (maybe `dev-`) for these branches so we can easily extend CI processes to include them